lib/perl/Genome/Library.t

```diff
@@ -2,37 +2,62 @@
 
 BEGIN {
     $ENV{UR_DBI_NO_COMMIT} = 1;
-    $ENV{UR_USE_DUMMY_AUTOGENERATED_IDS} = 1;
 }
 
 use strict;
 use warnings;
 
 use above "Genome";
 
-use Data::Dumper 'Dumper';
-use Test::More;
-
-use_ok('Genome::Library') or die;
-
-my $sample = Genome::Sample->create(name => '__TEST_SAMPLE__');
-ok($sample, 'create sample');
-my $library = Genome::Library->create(
-    sample => $sample,
-    name => $sample->name . "-extlibs",
-    original_insert_size => '1kb',
-    library_insert_size => '300-500',
-    protocol => 'karate chop',
-    transcript_strand => 'unstranded',
-);
-ok($library, 'create library');
-isa_ok($library, 'Genome::Library');
-ok($library->does('Genome::Role::Notable'), 'Library does Genome::Role::Notable');
-is($library->name, $sample->name . "-extlibs", "name is what is expected");
-
-ok($library->is_rna, 'is_rna true when transcript strand is set');
-
-my $commit = eval{ UR::Context->commit; };
-ok($commit, 'commit');
+use Test::More tests => 3;
+
+subtest 'use and roles' => sub{
+    plan tests => 2;
+
+    use_ok('Genome::Library') or die;
+    ok(Genome::Library->does('Genome::Role::Notable'), 'Library does Genome::Role::Notable');
+
+};
+
+my $library;
+subtest 'create' => sub{
+    plan tests => 4;
+
+    my $sample = Genome::Sample->create(name => '__TEST_SAMPLE__');
+    ok($sample, 'create sample');
+
+    $library = Genome::Library->create(
+        sample => $sample,
+        name => $sample->name . "-extlibs",
+        original_insert_size => '1kb',
+        library_insert_size => '300-500',
+        protocol => 'karate chop',
+    );
+    isa_ok($library, 'Genome::Library');
+    is($library->name, $sample->name . "-extlibs", "name is what is expected");
+
+    my $commit = eval{ UR::Context->commit; };
+    ok($commit, 'commit');
+
+};
+
+subtest 'is_rna' => sub{
+    plan tests => 4;
+
+    $library->sample->extraction_type('dna');
+    is($library->is_rna, 0, 'is NOT rna when extraction_type is dna and transcript_strand is UNDEF');
+
+    $library->transcript_strand('unstranded');
+    is($library->is_rna, 1, 'is rna when extraction_type is DNA and transcript_strand is unstranded');
```

Comment:
Probably not, I just added a test that preserved the logic as is in the method. The best place to validate this would be with an '__error__' to have transcript strand set and be 'DNA' on DB save.